src/compiler/moduleSpecifiers.ts

```diff
@@ -194,7 +194,12 @@ namespace ts.moduleSpecifiers {
         const symlinks = mapDefined(files, sf =>
             sf.resolvedModules && firstDefinedIterator(sf.resolvedModules.values(), res =>
                 res && res.resolvedFileName === importedFileName ? res.originalPath : undefined));
-        return symlinks.length === 0 ? getAllModulePathsUsingIndirectSymlinks(files, getNormalizedAbsolutePath(importedFileName, host.getCurrentDirectory ? host.getCurrentDirectory() : ""), getCanonicalFileName, host) : symlinks;
+        const cwd = host.getCurrentDirectory ? host.getCurrentDirectory() : "";

```

Comment:
I'm not sure I follow all the logic here, but there may be some duplicated effort. A few lines above this we get symlinks, then in `discoverProbableSymlinks` there is some nearly identical (copy-pasted?) code to do the same thing.

Reply:
As far as I know, it's not duplicated at all. Here we find files whose realpath differed from their actual path, _provided those files refer to the imported file_. In `discoverProbableSymlinks`, we crawl _all_ files looking for common parts among all files with symlinks to find probable directory junction links. If I remember correctly, anyway.